src/sql/src/session/metadata.rs

```diff
@@ -0,0 +1,65 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::fmt::Debug;
+
+use mz_adapter_types::connection::ConnectionId;
+use mz_repr::role_id::RoleId;
+
+use crate::plan::PlanContext;
+use crate::session::user::{RoleMetadata, User};
+use crate::session::vars::SessionVars;
+
+pub trait SessionMetadata: Debug + Sync {
```

Comment:
~~Instead of having a trait `SessionMetadata`, what do you think about a `Session` owning a `SessionMeta` struct, and then could return a `&SessionMeta` or easily clone it to hand out a reference-less copy?~~

Nevermind, it looks like `PlanContext` might make this hard!